packages/devtools/client/components/ServerTaskDetails.vue

```diff
@@ -0,0 +1,355 @@
+    inputs.filter(({ active, key, value }) => active && key && value !== undefined).map(({ key, value }) => [key, value]),
+  )
+  return Object.entries(formatted).length ? formatted : undefined
+}
+
+async function fetchData() {
+  started.value = true
+  fetching.value = true
+
+  const start = Date.now()
+
+  telemetry('server-tasks:run', {
+    method: method.value,
+  })
+
+  try {
+    response.data = await ($fetch as $Fetch)(finalURL.value, {
+      method: 'POST', // routeMethod.value.toUpperCase() as any,
+      headers: parsedHeader.value,
+      query: parsedQuery.value,
+      body: parsedBody.value,
+      onResponse({ response: res }) {
+        response.contentType = (res.headers.get('content-type') || '').toString().toLowerCase().trim()
+        response.statusCode = res.status
+        response.error = undefined
+      },
+      onResponseError(res) {
+        response.error = res.response._data
+        response.data = res.response._data
+      },
+    })
+  }
+  catch (err: any) {
+
+  }
+
+  fetching.value = false
+  response.fetchTime = Date.now() - start
+}
+
+const tabs = computed(() => {
+  const items = []
+  items.push({
+    name: 'Query',
+    slug: 'query',
+    length: routeInputs.query.length,
+  })
+  items.push({
+    name: 'Body',
+    slug: 'body',
+    length: routeInputs.body.length,
+  })
+  return items
+})
+
+watchEffect(() => {
+  if (selectedTabInput.value === 'json') {
+    if (typeof routeInputBodyJSON.value === 'string') {
+      try {
+        routeInputBodyJSON.value = JSON.parse(routeInputBodyJSON.value)
+      }
+      catch {}
+    }
+  }
+})
+
+const savedRouteInputs = useLocalStorage<{ task: string, tab: string, inputs: any }[]>('nuxt-devtools:server-tasks:inputs', () => [], {
+  window: window.parent,
+})
+
+watchDebounced([routeInputs, activeTab], () => {
+  const savedEntry = savedRouteInputs.value?.find(entry => entry.task === props.task.name)
+
+  if (!savedEntry) {
+    const newEntry = {
+      task: props.task.name,
+      tab: 'query',
+      inputs: routeInputs,
+    }
+    savedRouteInputs.value.push(newEntry)
+
+    if (!activeTab.value)
+      activeTab.value = newEntry.tab
+  }
+  else {
+    if (!activeTab.value)
+      activeTab.value = savedEntry.tab
+
+    if (savedEntry.tab !== activeTab.value)
+      savedEntry.tab = activeTab.value
+
+    //  update routeInputs with local storage
+    const { body, query, headers } = savedEntry.inputs
+    Object.assign(routeInputs, { body, query, headers })
+  }
+}, { immediate: true, deep: true, debounce: 500 })
+
+const copy = useCopy()
+</script>
+
+<template>
+  <div h-full w-full flex="~ col">
+    <div flex="~ col gap-2" flex-none p4 n-navbar-glass>
+      <div flex="~ gap2 items-center">
+        <div v-tooltip="`Method is ${method} as ${method === 'GET' ? 'no' : ''} json body is sent`">
+          <NButton
+            class="n-badge-base n-sm"
+            :class="getRequestMethodClass(method)" pointer-events-none font-mono
```

Comment:
I am a bit hestiated that we duplicate this playground, possible to reuse the one from server APIs?

Reply:
it would be possible to make it work, if we want to have tasks under `server-routes` we can add them under a `tasks` collection and have an option to hide them maybe

Reply:
That's something I considered while doing it, and I picked this path because of the features server routes offer that tasks don't support. By merging them I believed it would just mislead the users in the usage.

But I completely get the idea ! Ideally the Right panel should be configurable from the route component and opt-in or opt-out of the right panel's HTTP features (body input, headers for example)

Reply:
Even though it looks tempting to merge the two because it use the same code, I believe that they are two different things and it might evolve each in its way in the future. So I would personally keep it separated.

Reply:
@antfu Just so I know if I need to work more on this, what is the final decision regarding this ?